src/org/ods/component/Stage.groovy

```diff
@@ -4,18 +4,20 @@ class Stage {
   protected def script
   protected def context
   protected Map config
+  protected String componentId
 
   public final String STAGE_NAME = 'NOT SET'
 
   Stage(def script, IContext context, Map config) {
     this.script = script
     this.context = context
     this.config = config
+    componentId = config.componentId ?: context.componentId
```

Comment:
It took me a while to get why you are not reading from `config.componentId` everywhere, which I expected. TBH I think the usage would become clearer if all those stages would read directly from `config.componentId`. I would even do the setup in each constructor itself - to advertise that this option is available to the stage. Right now, I can change the `componentId` for every stage, even though some stages don't care and still read from `context.componentId`.

Reply:
well we use it later in the echo ... :(     `script.withStage(STAGE_NAME + ' (' + componentId + ')', context) `